integer/src/lib.rs

```diff
@@ -694,6 +696,22 @@ pub fn binomial<T: Integer + Clone>(mut n: T, k: T) -> T {
     r
 }
 
+/// Calculate the multinomial coefficient.
+///
+/// Panics if no integers are specified.
+pub fn multinomial<T: Integer + Clone>(k: &[T]) -> T
+    where for<'a> T: Add<&'a T, Output = T>
+{
+    assert!(k.len() > 0, "need at least one integer");
+    let mut r = T::one();
+    let mut p = T::zero();
+    for i in k {
```

Comment:
The additional branch is actually slower for `u64`. It might be worth it for bigints.

Without skipping zeros:
```
test bench_multinomial ... bench:         276 ns/iter (+/- 3)
```

Skipping zeros:
```
test bench_multinomial ... bench:         307 ns/iter (+/- 11)
```

I used this benchmark:
```rust
#[bench]
fn bench_multinomial(b: &mut Bencher) {
    b.iter(|| multinomial(&[91u64, 6, 3]));
}
```